rotkehlchen/db/upgrades/v34_v35.py

```diff
@@ -37,17 +38,23 @@ def _create_new_tables(cursor: 'DBCursor') -> None:
     """)
 
 
-def _rename_assets_identifiers(cursor: 'DBCursor') -> None:
+def _rename_assets_identifiers(write_cursor: 'DBCursor') -> None:
     """Version 1.26 includes the migration for the global db and the references to assets
-    need to be updated also in this database"""
-    cursor.execute('SELECT identifier FROM assets')
+    need to be updated also in this database.
+    We need to update the rows instead of just deleting and inserting because the rows are
+    referenced and this triggers an update in cascade.
+    """
+    write_cursor.execute('SELECT identifier FROM assets')
     old_id_to_new = {}
-    for (identifier,) in cursor:
+    for (identifier,) in write_cursor:
         if identifier.startswith(ETHEREUM_DIRECTIVE):
             old_id_to_new[identifier] = strethaddress_to_identifier(identifier[6:])
-
+        elif identifier in OTHER_EVM_CHAINS_ASSETS:
+            old_id_to_new[identifier] = OTHER_EVM_CHAINS_ASSETS[identifier]
     sqlite_tuples = [(new_id, old_id) for old_id, new_id in old_id_to_new.items()]
-    cursor.executemany('UPDATE OR IGNORE assets SET identifier=? WHERE identifier=?', sqlite_tuples)  # noqa: E501
+    # Make sure that the new ids don't exist already in the user db
```

Comment:
1. Yes, it will raise a unique error and fail during the upgrade. This is bad since the upgrade is failing and this is fatal for the app.
2. Yes this can happen. On way to make it happen is open the 1.25.2 app with the new globaldb and then the identifiers will be copied. Then if you open the database with the new release you will have this identifiers in the userdb.

Reply:
1. Oh by "current version" I meant the version after the changes in this pr. I meant that the deletion you introduced will cause a foreign key error if asset already exists and is stored not only in the `assets` table. Or am I wrong and there won't be any problems?
2. Is it possible? I mean won't [this check](https://github.com/rotki/rotki/blob/e68665fcb4f8a6455bab16595a8db75c7faa8b56/rotkehlchen/globaldb/handler.py#L66) prevent from doing it?

Reply:
When this migration happens means that db version is v34 and that means that the assets in here are only from the v2 schema of the globaldb since this is the first version where they are packaged and 2. indeed checks it